### PR TITLE
Dataset filtering support

### PR DESCRIPTION
**Pull request by mwray**
Added segment filtering support in addition to a sample transform function to epic_dataset.

### CODE REVIEW

File: epic_kitchens/dataset/epic_dataset.py [hunk 1]
```diff
@@ -90,17 +90,45 @@ def __repr__(self):
...
-        super().__init__(_class_count[class_type])
+        """
+
+        Parameters
+        ----------
+        gulp_path
+            Path to gulp directory containing the gulped EPIC RGB or flow frames
+        class_type
+            One of verb, noun, verb+noun, determines what label the segment returns
+        class_getter
```
Comment (willprice) [inline epic_kitchens/dataset/epic_dataset.py]: Swap ordering of docs to match signature (with_metadata comes before in sig)

File: epic_kitchens/dataset/epic_dataset.py [hunk 2]
```diff
@@ -90,17 +90,45 @@ def __repr__(self):
...
+            Optionally provide a callable that takes in the gulp dict representing the segment from which
+            you should return the class you wish the segment to have
+        with_metadata
+            When True the segments will yield a tuple (metadata, class) where the class is defined by the
+            class getter and the metadata is the raw dictionary stored in the gulp file.
+        segment_filter
+            Optionally provide a callable that takes a segment and returns True if you want to keep the
+            segment in the dataset, or False if you wish to exclude it.
+        sample_filter
```
Comment (willprice) [inline epic_kitchens/dataset/epic_dataset.py]: sample_transform, not filter.

File: epic_kitchens/dataset/epic_dataset.py [hunk 3]
```diff
@@ -90,17 +90,45 @@ def __repr__(self):
...
+            Optionally provide a sample transform function which takes a list of PIL images and transforms
+            each of them. This is applied on the list of frames upon loading the dataset.
```
Comment (willprice) [inline epic_kitchens/dataset/epic_dataset.py]: "This is applied on the frames just before returning from `load_frames`"

File: epic_kitchens/dataset/epic_dataset.py (hunk 1 above)
Comment (mwray) [reply epic_kitchens/dataset/epic_dataset.py]: Fixed in 52af03f

File: epic_kitchens/dataset/epic_dataset.py (hunk 2 above)
Comment (mwray) [reply epic_kitchens/dataset/epic_dataset.py]: Fixed in 52af03f

File: epic_kitchens/dataset/epic_dataset.py (hunk 3 above)
Comment (mwray) [reply epic_kitchens/dataset/epic_dataset.py]: Fixed in 52af03f

**Comment by willprice** [comment]
Feel free to squash and merge once tests have passed

**Pull request merged**